neon_mq_connector/consumers/select_consumer.py

```diff
@@ -213,6 +238,10 @@ def reconnect(self, wait_interval: int = 1):
 
     def join(self, timeout: Optional[float] = None) -> None:
         """Terminating consumer channel"""
-        if self.is_consumer_alive and self.is_consuming:
+        if self.is_consumer_alive:
             self._close_connection(mark_consumer_as_dead=True)
-        super().join(timeout=timeout)
+        LOG.info(f"Stopped consumer. Waiting up to {timeout}s for thread to terminate.")
+        try:
+            threading.Thread.join(self, timeout=timeout)
```

Comment:
I think its better to use `super().join(timeout=timeout)` to seamlessly pass the context to the thread's join implementation

Reply:
I think these are equivalent in practice, but the change is just leftover from my debugging so I'll revert it to minimize the diff